rust/ql/lib/codeql/rust/elements/internal/VariableImpl.qll

```diff
@@ -366,6 +387,29 @@ module Impl {
     override string getAPrimaryQlClass() { result = "VariableAccess" }
   }
 
+  /** Holds if `e` occurs in the LHS of a (compound) assignment. */
+  private predicate assignLhs(Expr e) {
+    exists(BinaryExpr be |
+      be.getOperatorName().regexpMatch(".*=") and
+      e = be.getLhs()
+    )
+    or
+    exists(Expr mid |
+      assignLhs(mid) and
+      getImmediateParent(e) = mid
+    )
+  }
+
+  /** A variable write. */
+  class VariableWriteAccess extends VariableAccess {
+    VariableWriteAccess() { assignLhs(this) }
+  }
+
+  /** A variable read. */
+  class VariableReadAccess extends VariableAccess {
+    VariableReadAccess() { not this instanceof VariableWriteAccess }
+  }
```

Comment:
Isn't a write in a compound assignment also a read?

Reply:
Yes

Reply:
(addressed)

Thanks.